tests/core/test_farmer_harvester_rpc.py

```diff
@@ -440,10 +440,12 @@ def add_plot_directories(prefix: str, count: int) -> List[Path]:
     await wait_for_plot_sync(receiver, last_sync_id)
 
     for page_size in [1, int(total_count / 2), total_count - 1, total_count, total_count + 1, 100]:
-        request = dataclasses.replace(request, page_size=uint32(page_size))
+        # we know that the concrete type of request has a page_size field, but
+        # the type is a Protocol that doesn't, so mypy is right in complaining
```

Comment:
I think this is an issue with our protocol and that should be worked on.  I'll see what I can figure out.

Reply:
I don't know how to express those `dataclass` properties as a Protocol. We already declare that we have *read only* fields with those names, but I suppose `replace()` will also need to be able to construct a new object with new values somehow